scipy/ndimage/tests/test_measurements.py

```diff
@@ -1026,190 +1175,216 @@ def test_extrema02():
+def test_center_of_mass06(xp):
+    expected = (0.5, 0.5)
+    input = np.asarray([[1, 2], [3, 1]], dtype=bool)
+    input = xp.asarray(input)
     output = ndimage.center_of_mass(input)
-    assert_array_almost_equal(output, expected)
+    assert output == expected
 
 
-def test_center_of_mass07():
-    labels = [1, 0]
-    expected = [0.5, 0.0]
-    input = np.array([[1, 2], [3, 1]], bool)
+def test_center_of_mass07(xp):
+    labels = xp.asarray([1, 0])
+    expected = (0.5, 0.0)
+    input = np.asarray([[1, 2], [3, 1]], dtype=bool)
+    input = xp.asarray(input)
     output = ndimage.center_of_mass(input, labels)
-    assert_array_almost_equal(output, expected)
+    assert output == expected
 
 
-def test_center_of_mass08():
-    labels = [1, 2]
-    expected = [0.5, 1.0]
-    input = np.array([[5, 2], [3, 1]], bool)
+def test_center_of_mass08(xp):
+    labels = xp.asarray([1, 2])
+    expected = (0.5, 1.0)
+    input = np.asarray([[5, 2], [3, 1]], dtype=bool)
+    input = xp.asarray(input)
     output = ndimage.center_of_mass(input, labels, 2)
-    assert_array_almost_equal(output, expected)
+    assert output == expected
 
 
-def test_center_of_mass09():
-    labels = [1, 2]
-    expected = [(0.5, 0.0), (0.5, 1.0)]
-    input = np.array([[1, 2], [1, 1]], bool)
-    output = ndimage.center_of_mass(input, labels, [1, 2])
-    assert_array_almost_equal(output, expected)
+def test_center_of_mass09(xp):
+    labels = xp.asarray((1, 2))
+    expected = xp.asarray([(0.5, 0.0), (0.5, 1.0)], dtype=xp.float64)
+    input = np.asarray([[1, 2], [1, 1]], dtype=bool)
+    input = xp.asarray(input)
+    output = ndimage.center_of_mass(input, labels, xp.asarray([1, 2]))
+    xp_assert_equal(xp.asarray(output), xp.asarray(expected))
 
 
-def test_histogram01():
-    expected = np.ones(10)
-    input = np.arange(10)
+def test_histogram01(xp):
+    expected = xp.ones(10)
+    input = xp.arange(10)
     output = ndimage.histogram(input, 0, 10, 10)
     assert_array_almost_equal(output, expected)
 
 
-def test_histogram02():
-    labels = [1, 1, 1, 1, 2, 2, 2, 2]
-    expected = [0, 2, 0, 1, 1]
-    input = np.array([1, 1, 3, 4, 3, 3, 3, 3])
+def test_histogram02(xp):
+    labels = xp.asarray([1, 1, 1, 1, 2, 2, 2, 2])
+    expected = xp.asarray([0, 2, 0, 1, 1])
+    input = xp.asarray([1, 1, 3, 4, 3, 3, 3, 3])
     output = ndimage.histogram(input, 0, 4, 5, labels, 1)
     assert_array_almost_equal(output, expected)
 
 
-def test_histogram03():
-    labels = [1, 0, 1, 1, 2, 2, 2, 2]
-    expected1 = [0, 1, 0, 1, 1]
-    expected2 = [0, 0, 0, 3, 0]
-    input = np.array([1, 1, 3, 4, 3, 5, 3, 3])
+@skip_xp_backends(np_only=True, reasons=['object arrays'])
+def test_histogram03(xp):
+    labels = xp.asarray([1, 0, 1, 1, 2, 2, 2, 2])
+    expected1 = xp.asarray([0, 1, 0, 1, 1])
+    expected2 = xp.asarray([0, 0, 0, 3, 0])
+    input = xp.asarray([1, 1, 3, 4, 3, 5, 3, 3])
+
     output = ndimage.histogram(input, 0, 4, 5, labels, (1, 2))
```

Comment:
is this due to the tuple `(1, 2)`? If so, I think this should be resolved in the implementation - the docs say `index` can be a _sequence of ints_, so we should aim for that to work.

Reply:
Possibly. Don't know TBH, am surprised by object arrays, too. Am not volunteering to fix the implementation though :-).

Reply:
Okay, my concern was a mishandling of the input tuple, but in fact the *result* is an object array of arrays. That is bizarre! Let's just leave this alone...

Reply:
Actually, no - since this is documented to accept a "sequence of ints" and return an "ndarray", we shoudn't be producing object arrays here which are incompatible with alternative backends. I'll mark this for follow-up, we might be in need of a deprecation cycle.